PR for SRVLOGIC-189: Added the installtion and uninstallation assemblies for Openshift Serverless Logic docs (#76488)
openshift/openshift-docs: modules/serverless-logic-install-web-console.adoc

// Module included in the following assemblies:
//
// * /serverless/install/install-serverless-operator.adoc

:_content-type: PROCEDURE
[id="serverless-logic-install-web-console_{context}"]
= Installing the {ServerlessLogicOperatorName} from the web console

> You can install the {ServerlessLogicOperatorName} from OperatorHub by using the {ocp-product-title} web console. Installing this Operator enables you to install and use serverless workflows.

Author: Yes, I can omit the "Installing this Operator enables you to install and use serverless workflows." sentence.

.Prerequisites

* You have cluster administrator permissions on {ocp-product-title}, or you have cluster or dedicated administrator permissions on {rosa-product-title} or {dedicated-product-title}.
* For {ocp-product-title}, your cluster has the Marketplace capability enabled or the Red Hat Operator catalog source configured manually.
* You have logged in to the  web console.

.Procedure

. In the web console, navigate to the *Operators* -> *OperatorHub* page.

. Scroll, or type the keyword *Serverless Logic* into the *Filter by keyword* box to find the {ServerlessLogicOperatorName}.

. Review the information about the Operator and click *Install*.

> . On the *Install Operator* page:

Reviewer: Lead-ins to lists should be complete sentences. Something along these lines maybe for this one?
```suggestion
. On the *Install Operator* page, ensure that the configuration uses the following settings:
```

Author: Understood and added

.. The *Installation Mode* is *All namespaces on the cluster (default)*. This mode installs the Operator in the default `openshift-serverless-logic` namespace to watch and be made available to all namespaces in the cluster.

.. The *Installed Namespace* is `openshift-serverless-logic`.

.. Select an *Update Channel*.
+
The *stable* channel is the default.

> .. Select the *stable* channel as the *Update Channel*. The *stable* channel will enable installation of the latest stable release of the {ServerlessLogicOperatorName}.

Reviewer: Can these be combined? looks like the same setting

.. Select *Automatic* or *Manual* approval strategy.

. Click *Install* to make the Operator available to the selected namespaces on this {ocp-product-title} cluster.

. From the *Catalog* -> *Operator Management* page, you can monitor the {ServerlessLogicOperatorName} subscription's installation and upgrade progress.

> .. If you selected a *Manual* approval strategy, the subscription's upgrade status will remain *Upgrading* until you review and approve its install plan. After approving on the *Install Plan* page, the subscription upgrade status moves to *Up to date*.
> 
> .. If you selected an *Automatic* approval strategy, the upgrade status should resolve to *Up to date* without intervention.

Reviewer: They will have one or the other, so bullets make sense. Consider putting Automatic first since it is mentioned first above (and is the easier path).
```suggestion
** If you selected a *Manual* approval strategy, the subscription's upgrade status will remain *Upgrading* until you review and approve its install plan. After approving on the *Install Plan* page, the subscription upgrade status moves to *Up to date*.

** If you selected an *Automatic* approval strategy, the upgrade status should resolve to *Up to date* without intervention.
```

.Verification

> After the Subscription's upgrade status is *Up to date*, select *Catalog* -> *Installed Operators* to verify that the {ServerlessLogicOperatorName} eventually shows up and its *Status* ultimately resolves to *InstallSucceeded* in the relevant namespace.
> 
> If it does not:
> 
> . Switch to the *Catalog* -> *Operator Management* page and inspect the *Operator Subscriptions* and *Install Plans* tabs for any failure or errors under *Status*.
> 
> . Check the logs in any pods in the `openshift-serverless-logic` project on the *Workloads* -> *Pods* page that are reporting issues to troubleshoot further.

Reviewer: Items under `.Verification` should be steps, looks like  first part is the actual verification step. The rest seems like troubleshooting, but could still go under the verification step:
```suggestion
* After the subscription's upgrade status is *Up to date*, select *Catalog* -> *Installed Operators* to verify that the {ServerlessLogicOperatorName} eventually shows up and its *Status* ultimately resolves to *InstallSucceeded* in the relevant namespace.
+
If it does not, check the following items:

.. Switch to the *Catalog* -> *Operator Management* page and inspect the *Operator Subscriptions* and *Install Plans* tabs for any failure or errors under *Status*.

.. Check the logs in any pods in the `openshift-serverless-logic` project on the *Workloads* -> *Pods* page that are reporting issues to troubleshoot further.
```

// Module included in the following assemblies:
//
// * /serverless/install/install-serverless-operator.adoc

:_content-type: PROCEDURE
[id="serverless-logic-install-web-console_{context}"]
= Installing the {ServerlessLogicOperatorName} from the web console

> You can install the {ServerlessLogicOperatorName} from the OperatorHub by using the {ocp-product-title} web console. Installing this Operator enables you to install and use Knative components.

Reviewer 2: ```suggestion
You can install the {ServerlessLogicOperatorName} from the OperatorHub by using the {ocp-product-title} web console. Installing this Operator enables you to install and use the serverless workflows.
```

> You can install the {ServerlessLogicOperatorName} from the OperatorHub by using the {ocp-product-title} web console. Installing this Operator enables you to install and use serverless workflows.

Reviewer 3: Not sure why, but this sounds better to me without "the". While "from Operator Hub" is more popular in the OCP docs, it seems that Serverless docs use "the". Your decision, of course. 

 ```suggestion
You can install the {ServerlessLogicOperatorName} from OperatorHub by using the {ocp-product-title} web console. Installing this Operator enables you to install and use serverless workflows.
```

Author: I will follow the OCP docs guidelines.

.Prerequisites

* You have cluster administrator permissions on {ocp-product-title}, or you have cluster or dedicated administrator permissions on {rosa-product-title} or {dedicated-product-title}.
* For {ocp-product-title}, your cluster has the Marketplace capability enabled or the Red Hat Operator catalog source configured manually.

> 

Reviewer 3: ```suggestion
```

Author: Fixed!

* You have logged in to the  web console.

.Procedure

. In the web console, navigate to the *Operators* -> *OperatorHub* page.

. Scroll, or type the keyword *Serverless Logic* into the *Filter by keyword* box to find the {ServerlessLogicOperatorName}.

. Review the information about the Operator and click *Install*.

. On the *Install Operator* page:

.. The *Installation Mode* is *All namespaces on the cluster (default)*. This mode installs the Operator in the default `openshift-serverless-logic` namespace to watch and be made available to all namespaces in the cluster.

.. The *Installed Namespace* is `openshift-serverless-logic`.

> .. Select an *Update Channel*.
> +
> * The *stable* channel enables installation of the latest stable release of the {ServerlessLogicOperatorName}. The *stable* channel is the default.
> 
> .. Select the *stable* channel as the *Update Channel*. The *stable* channel will enable installation of the latest stable release of the {ServerlessLogicOperatorName}.

Reviewer 3: Is this step repeated? I see the same format in the [Serverless Operator](https://docs.openshift.com/serverless/1.33/install/install-serverless-operator.html#serverless-install-web-console_install-serverless-operator) instructions.

```suggestion
.. Select an *Update Channel*.
+
.. Select the *stable* channel as the *Update Channel*. The *stable* channel will enable installation of the latest stable release of the {ServerlessLogicOperatorName}.
```
